tt_eager/tt_dnn/op_library/nlp_tms/nlp_tms.cpp

```diff
@@ -333,19 +333,21 @@ std::vector<Tensor> CreateQKVHeads::create_output_tensors(const std::vector<Tens
     //uint32_t num_q_heads_per_shard = q_shape[1] / num_w_cores;
 
     uint32_t q_shard_h = q_shape[0] * q_shape[1] * q_shape[2] / num_cores; // want the API to work for different sequence lengths
-    uint32_t kv_shard_h = k_shape[0] * k_shape[1] * k_shape[2] / num_cores; // want the API to work for different sequence lengths
-    auto q_spec = ShardSpec(all_cores, {q_shard_h, head_dim}, shard_orientation);
-    auto kv_spec = ShardSpec(all_cores, {kv_shard_h, head_dim}, shard_orientation);
+    uint32_t k_shard_h = k_shape[0] * k_shape[1] * k_shape[2] / num_cores; // want the API to work for different sequence lengths
+    uint32_t v_shard_h = v_shape[0] * v_shape[1] * v_shape[2] / num_cores; // want the API to work for different sequence lengths
 
+    auto q_spec = ShardSpec(all_cores, {q_shard_h, head_dim}, shard_orientation);
+    auto k_spec = ShardSpec(all_cores, {k_shard_h, transpose_k_heads ? input_shape[2] : head_dim}, shard_orientation);
+    auto v_spec = ShardSpec(all_cores, {v_shard_h, head_dim}, shard_orientation);
```

Comment:
We return the correct `k_shape` when we transpose anyways. All of these lines can just be:
```
auto q_spec = ShardSpec(all_cores, {q_shard_h, q_shard_h[-1]}, shard_orientation);
auto k_spec = ShardSpec(all_cores, {k_shard_h, k_shard_h[-1]}, shard_orientation);
auto v_spec = ShardSpec(all_cores, {v_shard_h, v_shard_h[-1]}, shard_orientation);
```

Reply:
Done